### PR TITLE
Fix for oneOf producing inconsistent casing

### PR DESCRIPTION
**Pull request by piq9117**
r/t https://github.com/scarf-sh/tie/issues/35

### CODE REVIEW

File: test/golden/test1.yaml.out [hunk 1]
```diff
@@ -549,39 +549,39 @@ import qualified Web.HttpApiData
 
 import Test.Schemas.PackageId
 
-newtype NISEValue = NISEValue
+newtype NiseValue = NiseValue
```
Comment (piq9117) [inline test/golden/test1.yaml.out:552]: :thinking: I don't think this is a desired change

**Comment by ken-scarf** [comment]
Ha! i just realized I'm using my civilian account for this.

File: src/Tie/Name.hs [hunk 2]
```diff
@@ -313,3 +314,9 @@ extractHaskellModule =
           xs ->
             [Text.intercalate "." xs]
    in concatMap extractModule . Text.words
+
+camelCaseIfNotAllCapitalized :: String -> String
+camelCaseIfNotAllCapitalized word =
+  if all (==True) $ isUpper <$> word
+  then word
+  else toCamelCase word
```
Comment (alexbiehl) [inline src/Tie/Name.hs:322]: Can you check if this has any effect at all? I can't quite pinpoint why the NISEValue would get into NiseValue earlier.

**Review by alexbiehl — COMMENTED** [review]
Thank you for your contribution! Though I have doubts on whether camelCaseIfNotAllCapitalized makes a difference at all.

File: src/Tie/Name.hs (hunk 2 above)
Comment (piq9117) [reply src/Tie/Name.hs:322]: You are right, sir. It was `camel` from `casing` package that I was using previously that caused that. `toCamelCase` is fine to use

**Pull request merged**